File: WorkoutApp.API/Models/Exercise.cs
using System.Collections.Generic;

namespace WorkoutApp.API.Models
{
    public class Exercise
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public Muscle PrimaryMuscle { get; set; }
        public Muscle SecondaryMuscle { get; set; }
        public List<UserFavoriteExercise> FavoritedBy { get; set; }
        public List<ExerciseStep> ExerciseSteps { get; set; }
        public List<EquipmentExercise> Equipment { get; set; }
        public List<ExerciseCategoryExercise> ExerciseCategorys { get; set; }
    }
}

File: WorkoutApp-SPA/src/app/_models/exercise.ts
import { Muscle } from './muscle';
import { Equipment } from './equipment';

export interface Exercise {
    id: number,
    name: string,
    primaryMuscle?: Muscle,
    secondaryMuscle?: Muscle,
    exerciseSteps?: ExerciseStep[],
    equipment?: Equipment[],
    exerciseCategorys?: ExerciseCategory[]
}

export interface ExerciseStep {
    exerciseStepNumber: number,
    description: string
}

export interface ExerciseCategory {
    id: number,
    name: string
}